src/App.test.js

```diff
@@ -33,3 +35,39 @@ test('food item is searched when button is clicked', () => {
   expect(screen.getByPlaceholderText('Search...')).toHaveValue('chicken');
   userEvent.type(input);
 });
+
+test('succesfully renders the login page button', () => {
+  render(<Login />);
+  const loginElement = screen.getByText('Login');
+  fireEvent.click(loginElement);
+});
+
+test('pass invalid password registration value to get error message', () => {
+  render(<Registration />);
+  const registerButton = screen.getByText('Register');
+  const username = screen.getByPlaceholderText('Username');
+  const password = screen.getByPlaceholderText('Password');
+
+  userEvent.type(username, 'Dragovich');
+  expect(screen.getByPlaceholderText('Username')).toHaveValue('Dragovich');
+  userEvent.type(password, '1');
+  expect(screen.getByPlaceholderText('Password')).toHaveValue('1');
+  expect(screen.queryByTestId('error-msg')).toBeInTheDocument();
+  expect(screen.queryByTestId('error-msg').textContent).toEqual('Use at least 4 characters');
+
+  fireEvent.click(registerButton);
+});
```

Comment:
I'm not certain, because the error message below doesn't seem to match the message sent by the server, but this test seems to expect server input validation. If that is the case, that's problematic because the server should not be required for client testing.

A better way to write a similar test may be to check for client-side invalidity on the HTML elements. See [checkValidity](https://developer.mozilla.org/en-US/docs/Web/API/ElementInternals/checkValidity) for details.

Reply:
Check out #42 for an example of this.